subsys/usb/device_next/class/usbd_cdc_ncm.c

```diff
@@ -0,0 +1,1282 @@
+				sys_le16_to_cpu(nth16->wNdpIndex), len);
+		return false;
+	}
+
+	if (sys_le16_to_cpu(nth16->wSequence) != 0
+	    &&  sys_le16_to_cpu(nth16->wSequence) != data->rx_sequence + 1) {
+		LOG_ERR("problem with sequence: %d %d", data->rx_sequence,
+				sys_le16_to_cpu(nth16->wSequence));
+	}
+	data->rx_sequence = sys_le16_to_cpu(nth16->wSequence);
+
+	/*
+	 * check (first) NDP(16)
+	 */
+	const struct ndp16_t *ndp16 = (const struct ndp16_t *)
+			(ntb->data + sys_le16_to_cpu(nth16->wNdpIndex));
+
+	if (sys_le16_to_cpu(ndp16->wLength) < sizeof(struct ndp16_t)
+			+ 2*sizeof(struct ndp16_datagram_t)) {
+		LOG_ERR("  ill ndp16 length: %d", sys_le16_to_cpu(ndp16->wLength));
+		return false;
+	}
+	if (sys_le32_to_cpu(ndp16->dwSignature) != NDP16_SIGNATURE_NCM0
+			&&  sys_le32_to_cpu(ndp16->dwSignature) != NDP16_SIGNATURE_NCM1) {
+		LOG_ERR("  ill signature: 0x%08x",
+				(unsigned int)sys_le32_to_cpu(ndp16->dwSignature));
+		return false;
+	}
+	if (sys_le16_to_cpu(ndp16->wNextNdpIndex) != 0) {
+		LOG_ERR("  cannot handle wNextNdpIndex!=0 (%d)",
+				sys_le16_to_cpu(ndp16->wNextNdpIndex));
+		return false;
+	}
+
+	const struct ndp16_datagram_t *ndp16_datagram = (const struct ndp16_datagram_t *)
+			(ntb->data + sys_le16_to_cpu(nth16->wNdpIndex) + sizeof(struct ndp16_t));
+	int ndx = 0;
+	uint16_t max_ndx = (uint16_t)((sys_le16_to_cpu(ndp16->wLength)
+			- sizeof(struct ndp16_t)) / sizeof(struct ndp16_datagram_t));
+
+	if (max_ndx > CFG_CDC_NCM_RCV_MAX_DATAGRAMS_PER_NTB + 1) {
+		/* number of datagrams in NTB > 1 */
+		LOG_ERR("<<xyx %d (%d)", max_ndx - 1, sys_le16_to_cpu(ntb->nth.wBlockLength));
+	}
+	if (sys_le16_to_cpu(ndp16_datagram[max_ndx-1].wDatagramIndex) != 0
+			||  sys_le16_to_cpu(ndp16_datagram[max_ndx-1].wDatagramLength) != 0) {
+		LOG_DBG("  max_ndx != 0");
+		return false;
+	}
+	while (sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramIndex) != 0
+			&&  sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramLength) != 0) {
+		LOG_DBG("  << %d %d", sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramIndex),
+				sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramLength));
+		if (sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramIndex) > len) {
+			LOG_ERR("(EE) ill start of datagram[%d]: %d (%d)",
+					ndx,
+					sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramIndex), len);
+			return false;
+		}
+		if (sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramIndex)
+				+ sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramLength) > len) {
+			LOG_ERR("(EE) ill end of datagram[%d]: %d (%d)", ndx,
+				sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramIndex)
+				+ sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramLength), len);
+			return false;
+		}
+		++ndx;
+	}
+
+	LOG_HEXDUMP_DBG(ntb->data, len, "NTB");
+
+	/* -> ntb contains a valid packet structure */
+	return true;
+}   /* _cdc_ncm_frame_ok */
+
+
+
+static int cdc_ncm_acl_out_cb(struct usbd_class_data *const c_data,
+		struct net_buf *const buf, const int err)
+/**
+ * Frame received from host!
+ */
```

Comment:
I did not find this in style guides.  In eclipse doing it this way is more convenient because the pop-up hover of a function then first shows function name incl parameters and then the explanation.  Handy at least for me.

If you convince me otherwise I will change it.

Reply:
I was looking at other Zephyr source file.

But it looks like the Linux kernel coding style does not say anything about it! https://kernel.org/doc/html/latest/process/coding-style.html

I might have mistaken with http://man.openbsd.org/style.9 :)

Not up to me to tell anyway!

Reply:
ok, so let's wait and see.  Compliance check is also no longer complaining